Sources/MapboxMaps/Location/Pucks/LocationPuckManager.swift

```diff
@@ -114,4 +130,14 @@ internal class LocationPuckManager: LocationConsumer {
             createPuck()
         }
     }
+
+    internal func changePuckBearingSource(to newPuckBearingSource: PuckBearingSource) {
+        guard var puck = self.puck else {
+            Log.warning(forMessage: "Puck must exist to change a PuckBearingSource", category: "Location")
+            return
+        }
```

Comment:
why do you want to do this? if we're storing the `newPuckBearingSource` below, then we can just use that to configure the puck if/when the puck is created right?

Reply:
Doing this will mean that the puckBearingSource must only be set after the style has finished loading, which is a bit of an arbitrary constraint.